crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -1004,6 +1050,44 @@ impl OwnUserIdentityData {
             })
             .collect()
     }
+
+    /// Was this identity previously verified, and is no longer?
+    ///
+    /// Such a violation should be reported to the local user by the
+    /// application, and resolved by
+    ///
+    /// - Verifying the new identity with
+    ///   [`OwnUserIdentity::request_verification`]
+    /// - Or by withdrawing the verification requirement
+    ///   [`OwnUserIdentity::withdraw_verification`].
+    pub fn has_verification_violation(&self) -> bool {
+        self.was_previously_verified() && !self.is_verified()
```

Comment:
A `matches!` macro here would be minutely more efficient, and maybe also easier to understand?

Reply:
so it would. Thanks.